x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/FrameGroupID.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.profiling;
+
+import java.util.Objects;
+
+public class FrameGroupID {
+    private static final char UNIX_PATH_SEPARATOR = '/';
+
+    public static String getBasenameAndParent(String fullPath) {
+        if (fullPath == null || fullPath.isEmpty()) {
```

Comment:
We could use `org.elasticsearch.common.Strings.isEmpty()` here (I overlooked that in my original implementation...).